api/tests/test_tasks_customer.py

```diff
@@ -0,0 +1,67 @@
+from django.test import TestCase
+from unittest.mock import patch, MagicMock
+
+from api.serializers import CustomerSerializer
+from api.tasks import check_customer_is_partner
+from sugarcrm.error import CrmAuthenticationError
+
+
+class TasksCustomerTests(TestCase):
```

Comment:
This class is only testing `CustomerSerializer` so it's a bit misleading for me to see references to Tasks both in the class and file names.

Reply:
Now I'm not sure. What's this class actually testing? `Customer.fetch` perhaps?

Reply:
The class and filenames I kind of based from your previous tests_tasks.py where testing the OrderSerializer too, but this one it is also mocking the .fetch to return an empty Customer roles=[] that will eventually assert as False. when checking if _check_customer_is_partner

Reply:
but if we're calling the method that we mocked a few lines above, what are we testing? If you think about it you only ensure `unittest.mock` works. Our won't get called at all.

Reply:
I agree with the test_failure method 
    @patch("api.tasks.Customer", autospec=True)
```
    def test_failure(self, mock_client):
        mock_client.fetch.side_effect = CrmAuthenticationError
        serializer = CustomerSerializer(data=self.data)
        self.assertTrue(serializer.is_valid())
        customer = serializer.validated_data
        with self.assertRaises(CrmAuthenticationError):
            mock_client.fetch(customer["email"])

```
is not checking anything since the error . I am replacing that and adding by new tests where not moking the fetch...